### PR TITLE
Ria 5126

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-5126

### Change description ###
RIA-5126 changes for WA Release2

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by tarun-palisetty — DISMISSED** [review]
LGTM

**Review by tarun-palisetty — APPROVED** [review]

**Review by tarun-palisetty — DISMISSED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/WaFieldsPublisher.java [hunk 1]
```diff
@@ -49,13 +49,17 @@ public void addLastModifiedApplication(
             String state,
             String applicantRole) {
 
+        System.out.println("Evaluate wa-R2-feature: " + featureToggler.getValue("wa-R2-feature", false));
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/WaFieldsPublisher.java:52]: @Lee365 these will need to be removed or replaced with proper Sl4j logging.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/WaFieldsPublisher.java [hunk 2]
```diff
@@ -49,13 +49,17 @@ public void addLastModifiedApplication(
...
+
         if (featureToggler.getValue("wa-R2-feature", false)) {
 
             final MakeAnApplication lastModifiedApplication = new MakeAnApplication(
                     applicant, type, details,
                     evidence, dateProvider.now().toString(), decision,
                     state);
 
+            System.out.println("lastModifiedApplication type: " + lastModifiedApplication.getType());
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/WaFieldsPublisher.java:61]: @Lee365 same as above.

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/payment/PayAndSubmitConfirmation.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ public PostSubmitCallbackResponse handle(
 
         boolean isException = false;
         try {
-            asylumCase = feePayment.aboutToSubmit(callback);
+            //asylumCase = feePayment.aboutToSubmit(callback);
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/payment/PayAndSubmitConfirmation.java:85]: @Lee365 not sure what the purpose of this try block is now?

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/MarkPaymentRequestSentConfirmation.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PostSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PostSubmitCallbackHandler;
+
+@Component
+public class MarkPaymentRequestSentConfirmation implements PostSubmitCallbackHandler<AsylumCase> {
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/MarkPaymentRequestSentConfirmation.java:13]: @Lee365 can this be placed behind the same `wa-R2-feature` feature flag?

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AddPaymentRequestCaseNoteHandler.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage.equals(ABOUT_TO_SUBMIT) && callback.getEvent().equals(MARK_PAYMENT_REQUEST_SENT);
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AddPaymentRequestCaseNoteHandler.java:53]: @Lee365 can this also be placed behind the same `wa-R2-feature` feature flag

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AddPaymentRequestCaseNoteHandler.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        Appender<CaseNote> caseNoteAppender,
+        DateProvider dateProvider,
+        UserDetails userDetails
+    ) {
+        this.caseNoteAppender = caseNoteAppender;
+        this.dateProvider = dateProvider;
+        this.userDetails = userDetails;
+    }
+
+    public boolean canHandle(
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AddPaymentRequestCaseNoteHandler.java:43]: @Lee365 can this handler be configured to be behind the `wa-R2-feature` flag

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/payment/PayAndSubmitConfirmation.java (hunk 3 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/payment/PayAndSubmitConfirmation.java:85]: This feepayment should not be commented - my mistake.
Please uncomment

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

**Pull request merged**